kong/router/atc.lua

```diff
@@ -877,7 +897,17 @@ function _M._set_ngx(mock_ngx)
 end
 
 
-_M.schema          = CACHED_SCHEMA
+-- for db schema validation
+function _M.schema(protocols)
+  local p = assert(protocols[1]:sub(1, 4))
+
+  if p == "http" or p == "grpc" then
```

Comment:
if protocols is `https`?

Reply:
Yes, Here we check the `sub(1, 4)`, which means `http` or `https`.

Reply:
Would maintaining a table for protocols and schemas be better?

e.g:
```lua 
local protocol_to_shcema = {
  http = HTTP_SCHEMA,
  https = HTTP_SCHEMA,
  grpc = HTTP_SCHEMA,
  grpcs = HTTP_SCHEMA,
  tcp = STREAM_SCHEMA,
  udp = STREAM_SCHEMA,
  ...
}

local schema = protocol_to_shcema(protocol)
```

I feel this way is more readable.

Reply:
done, thanks.